src/simularium/VisGeometry.ts

```diff
@@ -1129,35 +1192,38 @@ class VisGeometry {
     }
 
     /**
-     *   Map Type ID -> Geometry
+     * Map Type ID -> Geometry
      * This mapping is two level.
      * First agent type id --> meshName, via visGeomMap
      * Then meshName --> actual mesh, via meshRegistry (or pdbRegistry)
      */
     private mapIdToGeom(
         id: number,
-        displayType: string,
+        displayType: GeometryDisplayType,
         url?: string,
         color?: string
     ): void {
         this.logger.debug(`Geo for id ${id} set to '${url}'`);
         const unassignedName = `${VisAgent.UNASSIGNED_NAME_PREFIX}-${id}`;
-        const isMesh = displayType === "OBJ";
-        const isPDB = displayType === "PDB";
+        const isMesh = displayType === GeometryDisplayType.OBJ;
+        const isPDB = displayType === GeometryDisplayType.PDB;
         console.log(color); // TODO: handle color
         if (!url) {
             // displayType not either pdb or obj, will show a sphere
             // TODO: handle CUBE, GIZMO etc
             return;
         }
+        const urlOrPath = checkAndSanitizePath(url);
         this.visGeomMap.set(id, {
-            meshName: isMesh ? url : DEFAULT_MESH_NAME,
-            pdbName: isPDB ? url : "",
+            name: urlOrPath || DEFAULT_MESH_NAME,
+            displayType,
         });
-        if (isMesh) {
-            this.attemptToLoadGeometry(url, this.meshRegistry, "loadObj");
-        } else if (isPDB) {
-            this.attemptToLoadGeometry(url, this.pdbRegistry, "loadPdb");
+        if (isMesh || isPDB) {
+            this.attemptToLoadGeometry(
+                urlOrPath,
+                this.meshRegistry,
+                displayType
+            );
```

Comment:
@toloudis, i've been meaning to ask about this: In lines 1227 -1231, are these a remnant of assuming if a model is a pdb geo, it's always going to be a pdb model, and of the separate registries? my thinking is that anything that fails will default to a SPHERE mesh instead of a single atom pdb.

Reply:
umm.. yes? I think both things might be true but I'm sure we want to use sphere mesh for the default/error fallback rather than single atom pdb.  At least for now.

Reply:
it's going to be a bit weird having all this `PDBModel | MeshLoadRequest` all over the place and we might need an even more generic "delayed instanced object container" kind of thing that hides that.

Reply:
yeah, I guess that's my actual question. If a request fails, should we revert the type to the default geometry (same as if they didn't supply any geometry info at all), or should we try to be consistent with the displayType provided and keep this single atom pdb as the fallback to a failed pdb request?

Reply:
I'm not sure if this single atom pdb code is ever hit

Reply:
took a bit more time to read and think about this, and I think this is dead code now.  Everything having to do with UNASSIGNED_NAME_PREFIX (including in VisAgent.ts).

Reply:
thanks